src/bin/inx-chronicle/api/core/routes.rs

```diff
@@ -333,6 +333,13 @@ async fn commitment(
     Path(commitment_id): Path<SlotCommitmentId>,
     headers: HeaderMap,
 ) -> ApiResult<IotaRawResponse<SlotCommitment>> {
+    database
+        .collection::<CommittedSlotCollection>()
+        .get_id_for_slot_index(commitment_id.slot_index())
+        .await?
+        .and_then(|id| (id == commitment_id).then_some(()))
+        .ok_or(MissingError::NoResults)?;
+
     commitment_by_index(database, Path(commitment_id.slot_index()), headers).await
```

Comment:
rather than call the database twice just to avoid a small amount of repeated code, maybe we should just fully implement this method separately from `commitment_by_index`

Reply:
hmm, that does seem to require a new index though, right? :thinking:

Reply:
This is what I would do:
```rust
   let slot_commitment = database
        .collection::<CommittedSlotCollection>()
        .get_commitment(index)
        .await?
        .ok_or(MissingError::NoResults)?;

    if slot_commitment.commitment_id != commitment_id {
        return Err(MissingError::NoResults);
    }

    if matches!(headers.get(axum::http::header::ACCEPT), Some(header) if header == BYTE_CONTENT_HEADER) {
        return Ok(IotaRawResponse::Raw(slot_commitment.commitment.data()));
    }
    Ok(IotaRawResponse::Json(slot_commitment.commitment.into_inner()))
```

Reply:
ah, sure ... that is what I had done initially. I didn't like it because of the repitition and because we have the index on the slot index, but I don't mind.